src/katsdpcontroller/schemas/__init__.py

```diff
@@ -26,12 +27,12 @@
 _env = jinja2.Environment(loader=jinja2.PackageLoader(__name__, "."))
 
 
-def _normalise_version(version):
+def _normalise_version(version: Union["ComparableVersion", str]) -> "ComparableVersion":
```

Comment:
Can we not use the `|` syntax? Or are we still stuck on Python 3.8 for the controller?

Reply:
The controller still uses the katsdpdockerbase base image, which is Python 3.8.